Fix Issue #588. Only highlight runtime errors at the appropriate line…

jpgualdarrama opened the pull request:
… for file in which the error occurred.

I have verified that this pull request:

* [ ] has no linting errors (`npm run lint`)
* [ ] is from a uniquely-named feature branch and has been rebased on top of the latest master. (If I was asked to make more changes, I have made sure to rebase onto master then too)
* [ ] is descriptively named and links to an issue number, i.e. `Fixes #123`

catarak commented on client/modules/IDE/components/Editor.jsx:
```diff
@@ -197,7 +197,7 @@ class Editor extends React.Component {
             const n = consoleEvent.data[0].replace(')', '').split(' ');
             const lineNumber = parseInt(n[n.length - 1], 10) - 1;
-            if (!Number.isNaN(lineNumber)) {
+            if (!Number.isNaN(lineNumber) && `${consoleEvent.source}.js` === this.props.file.name) {
```
what if the error is in javascript that is in a script tag in the html file?

jpgualdarrama replied:
I'll investigate what I can do in that case, but from what I found so far, there may be other fixes required to correctly report errors from javascript in a script tag anyway. When a runtime error is found in a <script> tag in the html, the filename is not reported and the line number is not reported correctly. I made a sketch to demonstrate: https://editor.p5js.org/jon_gualdarrama/sketches/NXb-s08x1

jpgualdarrama replied:
Commit 39efd19 includes part of what I believe has to be done in order to highlight error lines in a <script> tag. It isn't complete yet, as the line number reported in the error will not match the line number in the html file on which the error actually occurred, so I set the line class on the correct line of the html file. After some googling, I've convinced myself that it has something to do with source maps - https://trackjs.com/blog/debugging-with-sourcemaps/. Will continue to investigate.

catarak replied:
i don't think it has to do with source maps—the sketch gets compiled into one long html file, which messes with the line numbers, and the necessary offsets are calculated by a function called `getAllScriptOffsets`, which is in `client/utils/consoleUtils.js`. looking at this function now, i see that it's only capturing the offsets for js files, and not inline js.

catarak commented:
another potential solution to this PR is to change the editor focus to the file that has the error—what do you think about this flow, versus just showing the error in the console if the file isn't selected?

jpgualdarrama commented:
I'm not sure how that would resolve this PR; would that help resolve the problem with the html files?

Also, I don't think just changing the focus to the be the file in which the error occurred would be sufficient to resolve the issue, as the problem is that once the error occurs on line _X_ of some file, no matter what file is currently active in the editor, line _X_ will always be highlighted, assuming the file has at least _X_ lines in it. If the user switches files while the error still exists, the same issue will continue to occur.

jpgualdarrama commented:
I think I've fixed the issue in `getAllScriptOffsets` to include index.html and updated `Editor.jsx` to highlight the correct line in index.html @catarak
